Fix merging locations that are ProjectAliases

mo-nathan opened the pull request:
To test:

Create a location that is a misspelling of some existing location.
Go to the project alias index page for a project (e.g., /projects/352/aliases).
Create an alias for the misspelled location.
Merge the misspelled location into the correctly spelled location (may require being in admin mode). Location merges happen when you change the name of a location to the name of another location.
On main the ProjectAlias will get deleted. On the branch it should get correctly reassigned to the correctly spelled
ProjectAlias for a location you want to merge (e.g., /locations/29279 should get merged into /locations/411).

mo-nathan committed:
Fix merging locations that are ProjectAliases

JoeCohen reviewed (commented):
I manually tested, but it didn't work/
- I created a misspelt Location (Cape Codd, Massachusetts, USA)
- For the NEMF Project, I created a Project Alias ("The Cape") for the misspelled location.
**Issue**: Is this intended: I went to (and refreshed) the Project Locations page. The misspelt Location is not listed.
- I merged the misspelled Location into the correctly spelled one.
Result: The alias disappeared.

Other: 
Is this intended: I didn't see a way for a user to get to the Project Aliases page.

mo-nathan commented:
Thanks for testing it out.  I had noticed that there isn't a way for users to get to the Project Aliases page.  The case I had tested previously actually involved there being an observation for the misspelled name.  For the name to show in the locations tab it is required that there be at least one observation associated with the project that is at that location.

mo-nathan commented:
I'm not seeing the alias disappearing from the alias index page when I tried your example.  It still isn't listed on the Locations tab, but that's expected if there are no associated observations.

mo-nathan committed:
- Add location link to location checklist
- Get location checklist to select the location tab
- Access to the ProjectAlias index when there are no observations
- Fix location name order on the ProjectAlias index page

mo-nathan commented:
@JoeCohen I added a "Project Aliases" tab to the Project page.  I also fixed an issue with location name formating on the Project Alias index page, added a link to the location when on a checklist for a location, and made it soe the "Locations" tab is selected when viewing a location checklist for a project.  Whew!  (Still need to write some tests...)

JoeCohen commented:
I intended to retry project-location-merge.
But — after updating my local db to the newest snapshot, pulling main, and migrating — your branch doesn’t work on my local.  
If you like, I will revert the db & latest migration (I think that's what I need to do to make your branch work) and try again.

mo-nathan committed:
Merge branch 'main' into njw-project-alias-location-merge

mo-nathan commented:
Updated to work against current main.

JoeCohen commented:
It works as expected. IDK what happened yesterday.

mo-nathan committed:
Merge branch 'main' into njw-project-alias-location-merge

mo-nathan committed:
Test that the Project Aliases tab is selected and that locations
respect the user's location format.

mo-nathan committed:
Ensure that location checklist mentioned location and selects the
location tab in the context of a project.

mo-nathan committed:
Rubocop
